File: api/save_to_db.py
import os
import json
import psycopg2
from http.server import BaseHTTPRequestHandler

class handler(BaseHTTPRequestHandler):
    def _cors(self):
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.send_header('Access-Control-Allow-Methods', 'GET,POST,OPTIONS')

    def do_OPTIONS(self):
        self.send_response(204)
        self._cors()
        self.end_headers()

    def do_POST(self):
        try:
            length = int(self.headers.get('Content-Length', '0'))
            body = self.rfile.read(length)
            data = json.loads(body.decode('utf-8') or '{}')

            conn = psycopg2.connect(os.environ.get('DATABASE_URL'))
            cur = conn.cursor()

            enrollment_no = data.get('enrollment_no', '')
            name = data.get('name') or data.get('student_name') or data.get('parent_name') or ''
            category = data.get('category', 'Student')
            phone = data.get('phone') or data.get('parent_phone') or ''
            employee_no = data.get('employee_no') or ''
            last_class = data.get('last_class') or ''
            json_data = json.dumps(data)

            conditions = []
            params = []
            if enrollment_no:
                conditions.append("enrollment_no = %s"); params.append(enrollment_no)
            if phone:
                conditions.append("phone = %s"); params.append(phone)
            if employee_no:
                conditions.append("data->>'employee_no' = %s"); params.append(employee_no)
            if last_class and name:
                conditions.append("(data->>'last_class' = %s AND name = %s)"); params.extend([last_class, name])
            duplicate = False
            if conditions:
                q = "SELECT id FROM registrations WHERE " + " OR ".join(conditions) + " LIMIT 1"
                cur.execute(q, params)
                row = cur.fetchone()
                duplicate = bool(row)
            if duplicate:
                self.send_response(409)
                self._cors()
                self.send_header('Content-Type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({'success': False, 'message': 'duplicate'}).encode('utf-8'))
                cur.close(); conn.close()
                return

            cur.execute(
                'INSERT INTO registrations (enrollment_no, name, category, phone, data, created_at) VALUES (%s, %s, %s, %s, %s, NOW()) RETURNING id',
                (enrollment_no, name, category, phone, json_data)
            )
            new_id = cur.fetchone()[0]
            conn.commit()
            cur.close(); conn.close()

            self.send_response(200)
            self._cors()
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'success': True, 'id': new_id, 'source': 'vercel-python'}).encode('utf-8'))
        except Exception as e:
            self.send_response(500)
            self._cors()
            self.send_header('Content-Type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({'success': False, 'error': str(e)}).encode('utf-8'))

File: netlify/functions/save_to_db.js
const { Client } = require('pg');
const mysql = require('mysql2/promise');

exports.handler = async (event) => {
  const headers = {
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Headers': 'Content-Type',
    'Access-Control-Allow-Methods': 'POST, OPTIONS'
  };

  if (event.httpMethod === 'OPTIONS') {
    return { statusCode: 204, headers };
  }

  if (event.httpMethod !== 'POST') {
    return { statusCode: 405, headers, body: 'Method Not Allowed' };
  }

  try {
    const data = JSON.parse(event.body);
    
    // 1. Try PostgreSQL if configured
    if (process.env.DATABASE_URL) {
      const client = new Client({
        connectionString: process.env.DATABASE_URL,
        ssl: { rejectUnauthorized: false }
      });
      await client.connect();
      
      const query = `
        INSERT INTO registrations (enrollment_no, name, category, phone, data, created_at)
        VALUES ($1, $2, $3, $4, $5, NOW())
        RETURNING id
      `;
      const values = [
        data.enrollment_no || '',
        data.name || data.student_name || data.parent_name || '',
        data.category || 'Student',
        data.phone || data.parent_phone || '',
        JSON.stringify(data)
      ];
      
      await client.query(query, values);
      await client.end();
      return { statusCode: 200, headers, body: JSON.stringify({ success: true, source: 'postgres' }) };
    }

    // 2. Try MySQL if configured
    if (process.env.MYSQL_URL) {
      const connection = await mysql.createConnection(process.env.MYSQL_URL);
      const [rows] = await connection.execute(
        'INSERT INTO registrations (enrollment_no, name, category, phone, data, created_at) VALUES (?, ?, ?, ?, ?, NOW())',
        [
          data.enrollment_no || '',
          data.name || data.student_name || data.parent_name || '',
          data.category || 'Student',
          data.phone || data.parent_phone || '',
          JSON.stringify(data)
        ]
      );
      await connection.end();
      return { statusCode: 200, headers, body: JSON.stringify({ success: true, source: 'mysql' }) };
    }

    return { 
      statusCode: 200, 
      headers, 
      body: JSON.stringify({ 
        success: false, 
        message: 'No database configured. Please set DATABASE_URL (Postgres) or MYSQL_URL (MySQL) in Netlify.' 
      }) 
    };

  } catch (error) {
    console.error('Database Error:', error);
    return { 
      statusCode: 500, 
      headers, 
      body: JSON.stringify({ success: false, error: error.message }) 
    };
  }
};
